core/src/main/java/io/grpc/internal/MessageFramer.java

```diff
@@ -342,7 +340,7 @@ private BufferChainOutputStream() {
 
     @Override
     public void write(int b) throws IOException {
-      singleByte[0] = (byte) b;
+      byte[] singleByte = new byte[]{(byte)b};
```

Comment:
Maybe?  It's immediately written to the buf below.  I can't imagine calling the single byte version being very fast anyway.  I expect this 1 byte array to stay in eden.  Lazy init is still not good, since it costs and extra pointer in the writer which will have to be traversed.  I considered a weakref to the byte array and init it that way, but that seems slower for something I expect the jvm to stack allocate.